### PR TITLE
Allow update a dependency to fixed version from tree

### PR DESCRIPTION
**Pull request by Or-Geva**
- [x] All [tests](https://github.com/jfrog/jfrog-vscode-extension#building-and-testing-the-sources) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `npm run format` for formatting the code before submitting the pull request.
-----
**What is it?** 
 * Allow update a dependency to fixed version from tree

**How to use it?**
 * Go to:  JFrog-Extention -> Tree -> Chose a node which includes fixed version -> Click on the arrow-up.
 
 **Supported Languages**
  * NPM
  * Maven
  * Go

### CODE REVIEW

**Review by yahavi — CHANGES_REQUESTED** [review]
Looks good! Please consider the following:
1. When an error occurred (i.e. when the dependency doesn't exist), focus on the log and/or toast an error.
2. I'd like to make the process more visible to the users. Please focus on the relevant location in the descriptor file, i.e. the <version> of the dependency in the pom.xml. Same as the eye button does, but focus on the version. 
3. Since the process runs an external process, please consider also adding a progress bar.

I'll review you code afterward. Thanks!

**Comment by Or-Geva** [comment]
The following improvements were added:
  1. When an error occurred, focus on the log and/or toast an error.
  2. Focus on the relevant location in the descriptor file if the dependency updated.
  3. Add a progress bar while updating dependency

File: src/main/DependencyUpdate/goDependencyUpdate.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export class GoUpdateDependency extends AbstractUpdateDependency {
+    constructor() {
+        super(GoUtils.PKG_TYPE);
+    }
+
+    /** @override */
+    public updateDependencyVersion(dependenciesTreeNode: DependenciesTreeNode, fixedVersion: string): void {
+        if (!(dependenciesTreeNode.parent instanceof GoTreeNode)) {
+            return;
+        }
```
Comment (yahavi) [inline src/main/DependencyUpdate/goDependencyUpdate.ts:16]: Please put all preconditions like this in an overriding `isMatched` function. Same for npm and Maven

**Review by yahavi — CHANGES_REQUESTED** [review]

File: src/main/DependencyUpdate/updateDependencyManager.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     * a quick  pick will appear to the user and a list of versions will be shown, the chosen version will be returned.
+     */
+    public async getFixedVersion(dependenciesTreeNode: DependenciesTreeNode) {
+        let fixedVersions: Collections.Set<string> = new Collections.Set<string>();
+        dependenciesTreeNode.issues.forEach(issue => {
+            if (issue.component === dependenciesTreeNode.componentId) {
+                issue.fixedVersions.forEach(fixedVersion => fixedVersions.add(fixedVersion));
+            }
+        });
+        let uniqueFixedVersions: string[] = fixedVersions.toArray();
```
Comment (yahavi) [inline src/main/DependencyUpdate/updateDependencyManager.ts:41]:
```suggestion
        let fixedVersionsArr: string[] = fixedVersions.toArray();
```

File: src/main/utils/mavenUtils.ts [hunk 3]
```diff
@@ -216,7 +230,7 @@ export class MavenUtils {
             node.contextValue = '';
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/utils/mavenUtils.ts:230]: Yea!.

File: src/main/commands/commandManager.ts [hunk 4]
```diff
@@ -38,15 +42,36 @@ export class CommandManager implements ExtensionComponent {
...
+        await this._focusManager.focusOnDependency(dependenciesTreeNode, FocusType.Dependency);
         this.onSelectNode(dependenciesTreeNode);
     }
 
     private doExcludeDependency(dependenciesTreeNode: DependenciesTreeNode) {
         this._exclusionManager.excludeDependency(dependenciesTreeNode);
     }
 
+    private async doUpdateDependencyVersion(dependenciesTreeNode: DependenciesTreeNode) {
+        await vscode.window.withProgress(
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply src/main/commands/commandManager.ts:55]: its title doesn't fit

**Review by yahavi — APPROVED** [review]

**Pull request merged**